### PR TITLE
Modernize test coreRfeature

### PR DESCRIPTION
**Pull request by perrydv**
This modernizes `test_coreRfeature` as follows:

* Everything is wrapped in `test_that`
* All `try(test_that(…))` idioms removed.
* Remaining `try` for `compileNimble` uses `silent=TRUE` and does lead to a managed use of `expect_equal` to check if compiler failure or success was as expected.
* Uses in `test-coreRfeature`, `test-allocation` and test-numericTypes` work on my machine.

### CODE REVIEW

File: packages/nimble/inst/tests/test_utils.R [hunk 1]
```diff
@@ -51,22 +51,28 @@ indexNames <- function(x) {
...
+    })
+}
+
+test_coreRfeature_internal <- function(input, verbose = TRUE, dirName = NULL) { ## a lot like test_math but a bit more flexible
   if(verbose) cat("### Testing", input$name, "###\n")
   runFun <- gen_runFunCore(input)
   nfR <- nimbleFunction(run = runFun)
-  nfC <- try(compileNimble(nfR, dirName = dirName))
+  ## This try is safe because failure is caught by expect_equal below
+  nfC <- try(compileNimble(nfR, dirName = dirName), silent = TRUE)
```
Comment (fritzo) [inline packages/nimble/inst/tests/test_utils.R]: This looks reasonable. I'm not sure it's appropriate here, but I've also found `expec_error` helpful in `test_size()`:
```r
if(failureExpected) {
  expect_error(compileNimble(...))
} else {
  nf <- compileNimble(...)
}
```

**Comment by fritzo** [comment]
LGTM

BTW Have you checked with both `test_package`/`test_file` and `Rscript`?

**Comment by perrydv** [comment]
That `expect_error` example would be cleaner but equivalent in behavior.  I'll try it.  Thanks for the reminder to try Rscript mode as well.   I ran via `test_that` locally and I guess they are running via Rscript on travis.

**Comment by fritzo** [comment]
@perrydv It looks like this merge conflicted with my travis-osx merge, possibly due to different behavior of `expect_identical` on linux and OS X. The current `devel` is now failing OS X but passing Linux. As you suggested, I think this should be an easy fix.

**Pull request merged**